src/shell.c

```diff
@@ -165,19 +165,40 @@ int main() {
 
         int argc = parse(input, command, args);
 
-        pid_t pid = vfork();
-
-        if (pid < 0) {
-            perror("fork");
-            exit(-1);
-        } else if (pid == 0) {
-            exec(command, args, argc);
-            exit(0);
-        } else {
-            wait(NULL);
+        if (strcmp(command, "cd") == 0) {
+            if (argc != 2) 
+                printf("Couldn't change directory\n");
+
+            else {
+                if (chdir(args[1]) == -1) 
+                    printf("Error from chdir\n");
+                        
+                else {
```

Comment:
We don't need this else logic, because we can see how the path changes in the terminal.